src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -717,7 +718,18 @@ internal async Task MergeResourcesWrapperAsync(long transactionId, bool singleTr
             new TokenStringCompositeSearchParamListTableValuedParameterDefinition("@TokenStringCompositeSearchParams").AddParameter(cmd.Parameters, new TokenStringCompositeSearchParamListRowGenerator(_model, new TokenSearchParamListRowGenerator(_model, _searchParameterTypeMap), new StringSearchParamListRowGenerator(_model, _searchParameterTypeMap), _searchParameterTypeMap).GenerateRows(mergeWrappers));
             new TokenNumberNumberCompositeSearchParamListTableValuedParameterDefinition("@TokenNumberNumberCompositeSearchParams").AddParameter(cmd.Parameters, new TokenNumberNumberCompositeSearchParamListRowGenerator(_model, new TokenSearchParamListRowGenerator(_model, _searchParameterTypeMap), new NumberSearchParamListRowGenerator(_model, _searchParameterTypeMap), _searchParameterTypeMap).GenerateRows(mergeWrappers));
             cmd.CommandTimeout = 300 + (int)(3600.0 / 10000 * (timeoutRetries + 1) * mergeWrappers.Count);
-            await cmd.ExecuteNonQueryAsync(cancellationToken);
+
+            if (enlistInTransaction)
```

Comment:
What is this doing differently than just passing false to ObtainSqlConnectionWrapperAsync?

Reply:
ObtainSqlConnectionWrapperAsync does not allow setting app name, because it was too difficult to make appropriate change in shared.

In general, enlisting in C# transaction is evil, so we should stay away from using it in all cases we can, and eventually get rid of it.